src/hooks/Circles/utils/optimisticallyDeletePosts.ts

```diff
@@ -54,10 +55,28 @@ const removeEdgeById = (currentData: InfinitePostsData, id: string) =>
 const removeRepliesById = (
   currentData: PostDetailsPostQueryResponse,
   id: string,
+  depth: number = 7, // set some default max depth
 ) => {
-  const deleted = remove(
-    currentData?.post.replies.edges,
-    (edge) => edge.node.id === id,
-  );
-  currentData.post.replyCount -= deleted.length;
+  let count = 0;
+  const recursiveRemove = (post: Post, currentDepth: number) => {
+    if (currentDepth > depth) {
+      return;
+    }
+
+    const deleted = remove(post.replies.edges, (edge) => edge.node.id === id);
+    post.replyCount -= deleted.length;
+
+    // Recurse into the replies of the remaining edges
+    post.replies.edges.forEach((edge) => {
+      if (edge.node.replies && edge.node.replies.edges) {
+        recursiveRemove(edge.node, currentDepth + 1);
+      }
+    });
+  };
+
+  if (currentData?.post.replies.edges) {
+    recursiveRemove(currentData.post, 0);
+  }
+
+  currentData.post.replyCount -= count;
```

Comment:
How does `count` get incremented? I am not seeing it.

Reply:
That was an artifact of something I had tried earlier, no longer present in this latest commit/refactor.